source/postprocess/mobility_statistics.cc

```diff
@@ -0,0 +1,201 @@
+                                                   fe_values.JxW(q));
+              }
+          }
+
+      const double global_velocity_square_integral
+        = Utilities::MPI::sum (local_velocity_square_integral, this->get_mpi_communicator());
+
+      const double vrms = std::sqrt(global_velocity_square_integral) /
+                          std::sqrt(this->get_volume());
+
+      //now compute surface vrms
+      // create a quadrature formula for the velocity.
+      const QGauss<dim-1> quadrature_formula_for_surface (this->introspection().polynomial_degree.velocities+1);
+  
+      FEFaceValues<dim> fe_face_values (this->get_mapping(),
+                                        this->get_fe(),
+                                        quadrature_formula_for_surface,
+                                        update_values |
+                                        update_JxW_values |
+                                        update_quadrature_points);
+      
+      std::vector<Tensor<1,dim>> velocities (fe_face_values.n_quadrature_points);
+      
+      std::map<types::boundary_id, double> local_velocity_square_integral_surface;
+      std::map<types::boundary_id, double> local_boundary_area;
+
+      const std::set<types::boundary_id>
+      boundary_indicators
+        = this->get_geometry_model().get_used_boundary_indicators ();
+      const types::boundary_id top_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+ 
+      // for every surface face that is part of a geometry boundary
+      // and that is owned by this processor,
+      // compute the squared*area velocity magnitude,
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned())
+          for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+            if (cell->face(f)->boundary_id() == top_boundary_id)
+              {
+                fe_face_values.reinit (cell, f);
+                // extract velocities
+                fe_face_values[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
+                velocities);
+                // determine the squared velocity on the face 
+                double local_sqvel = 0.0;
+                double local_fe_face_area = 0.0;
+                for (unsigned int q=0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    const double vel_mag = velocities[q].norm();
+                    local_sqvel += ((vel_mag * vel_mag) * fe_face_values.JxW(q));
+                    local_fe_face_area += fe_face_values.JxW(q);
+                  }
+                // then merge them with the squared velocities
+                // we found for other faces with the same boundary indicator
+                const types::boundary_id boundary_indicator
+                  = cell->face(f)->boundary_id();
```

Comment:
No need to recompute the boundary indicator. Look at line 101: You only get here if the boundary indicator is the top boundary. If you make the change from maps to doubles that I mentioned above the lines below will not need the boundary id at all.